compiler/luci/pass/src/DynamicBatchToSingleBatchPass.cpp

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/DynamicBatchToSingleBatchPass.h"
+
+#include <luci/IR/CircleNode.h>
+#include <loco.h>
+
+namespace luci
+{
+
+bool DynamicBatchToSingleBatchPass::run(loco::Graph *g)
+{
+  assert(g); // FIX CALLER UNLESS
+
+  bool changed = false;
+
+  auto graph_inputs = g->inputs();
+
+  // Assume the first dimension is batch dimension
+  const uint32_t BATCH_DIM = 0;
+
+  for (auto node : loco::input_nodes(g))
+  {
+    auto input_node = loco::must_cast<luci::CircleInput *>(node);
+
+    if (input_node->rank() == 0)
+      continue;
+
+    // Skip if batch dimension is known
+    if (input_node->dim(BATCH_DIM).known())
+      continue;
+
+    if (input_node->rank() != 4)
+    {
+      // Limit use only for rank 4 inputs (for NHWC and NCHW)
+      // TODO Enable this if necessary
+      throw std::runtime_error("First dimension of input is unknown, but its rank is not 4.");
+    }
```

Comment:
@seanshpark Now this pass throws an exception for non-rank 4 cases.

Reply:
Can you please add for another `rank_1_NEG` test with rank 1 input and expect throw?

Reply:
later when we are sure rank 1 works then we can remove this new test.
maybe a note about this would help future task.